### PR TITLE
Explain vendor vs proxy

### LINKED ISSUE #1338: docs: make it easy to compare vendoring and athens
**Is your feature request related to a problem? Please describe.**
My team found it difficult to compare and contrast vendoring with Athens.

**Describe the solution you'd like**
Documentation to make it easy to compare vendoring with go module proxies such as Athens.
Ideally the documentation would make it easy to compare the solutions on the users repo, with commands to copy and run to see the benefits of each solution (e.g. time and space considerations)
You could also consider explaining how to use the GOPRIVATE env var in this document.

**Describe alternatives you've considered**
A repo with examples of each solution (vendoring, go module proxy, neither) to compare and contrast each.

**Additional context**
My team had flakey builds as a result of errors resolving go modules.

### PR DESCRIPTION
**Pull request by schafer14**
**What is the problem I am trying to address?**

There have been a few questions in the recent past about when/why to use a
vendor directory instead of (or along with) Athens or another proxy.

**How is the fix applied?**

I added a section to the FAQ that explains some pros and cons for when to use
vendor and when to use a proxy server like Athens.

**Mention the issue number it fixes or add the details of the changes if it doesn't have a specific issue.**

Fixes #1338

This is following up on PR #1340

### CODE REVIEW

File: docs/content/faq.md [hunk 1]
```diff
@@ -87,3 +87,27 @@ Which currently includes:
 - hg
 - bzr
 - fossil
+
+### When should I use a vendor directory, and when should I use Athens?
+
+The Go community has used vendor directories for a long time before module proxies like Athens came along, so naturally each group collaborating on code should decide for themselves whether they want to use a vendor directory, use Athens, or do both!
+
+Vendor (without a proxy) are valuable when:
```
Comment (arschles) [inline docs/content/faq.md]: Can you write this as `Using a vendor directory (without a proxy) is valuable when:`?

File: docs/content/faq.md [hunk 2]
```diff
@@ -87,3 +87,27 @@ Which currently includes:
...
+
+- CI/CD systems don't have access to an Athens (even if it's internal)
+- When the vendor directory is so small that it is still faster to check it out from a repo than it is to pull zip files from the server
+- If you're coming from glide/dep or another dependency management system that leveraged the vendor directory
+
+Athens valuable when:
```
Comment (arschles) [inline docs/content/faq.md]: Can you write this as `Athens (without a vendor directory) is valuable when:`? I want the docs to be clear that we're talking about just a proxy, and deleting the entire vendor directory

File: docs/content/faq.md [hunk 3]
```diff
@@ -87,3 +87,27 @@ Which currently includes:
...
+
+- You have a new project
+- Your team requires that you use Athens (i.e. for isolation or dependency auditing)
+- Your vendor directory is large and causing slow checkouts and downloading from Athens speeds the build up
+  - For developers slow checkouts will not be as much of a problem as for ci tools which frequently need to checkout fresh copies of the project
+- You do not want to pollute your project with the vendor directory
+  - Extra noise in pull requests
+  - Extra difficulty doing fuzzy file searching in your project
```
Comment (arschles) [inline docs/content/faq.md]: Can you elaborate on this and the previous bullet point? It sounds like Athens will be valuable in reducing these two things. Is that right?

Comment (schafer14) [reply docs/content/faq.md]: This is awkward phrasing. Is this easier to follow?
```
- You want to remove the vendor directory from your project to:
  - Reduce noise in pull requests
  - Reduce difficulty doing fuzzy file searching in your project
```

Comment (arschles) [reply docs/content/faq.md]: Most definitely, that's easier to follow!

**Pull request merged**